pkg/operators/cli/clioperator.go

```diff
@@ -218,10 +225,64 @@ func (o *cliOperatorInstance) PreStart(gadgetCtx operators.GadgetContext) error
 			}
 
 			formatter.SetEventCallback(func(s string) {
-				fmt.Print(s)
-				fmt.Print("\n")
+				fmt.Println(s)
 			})
 
+			printHeader := func() {
+				fmt.Println(formatter.FormatHeader())
+			}
+
+			gadgetCtx.Logger().Infof("%+v", ds.Annotations())
+
+			headerFuncs := []func(){}
+			if ds.Annotations()[AnnotationClearScreenBefore] == "true" {
+				isTerminal := term.IsTerminal(int(os.Stdout.Fd()))
+				if isTerminal {
+					headerFuncs = append(headerFuncs, clearScreen)
+				}
+			}
+			headerFuncs = append(headerFuncs, printHeader)
+
+			for _, headerFunc := range headerFuncs {
+				headerFunc()
+			}
+
+			switch ds.Type() {
+			case datasource.TypeSingle:
```

Comment:
It is necessary to manage differently Single and Array to be able to clean up screen only after the whole array was printed and not after each element?

Reply:
I think it's necessary to know when there's a new array coming in to clear the screen, yes. Another solution would be to use Subscribe() for the data and SubscribeArray() (Priority-1) only to clear the screen (and ignore the data coming from it).